pkg/version/testhelpers/testhelpers.go

```diff
@@ -42,114 +41,63 @@ func run(cmd *exec.Cmd) error {
 	return nil
 }
 
-func goBuildEnviron(gopath string) []string {
-	environ := os.Environ()
-	for i, kvp := range environ {
-		if strings.HasPrefix(kvp, "GOPATH=") {
-			environ[i] = "GOPATH=" + gopath
-			return environ
+// unset GOPATH if it's set, so we use modules
+func goBuildEnviron() []string {
+	out := []string{}
+	for _, kvp := range os.Environ() {
+		if !strings.HasPrefix(kvp, "GOPATH=") {
+			out = append(out, kvp)
 		}
 	}
-	environ = append(environ, "GOPATH="+gopath)
-	return environ
+	return out
 }
 
-func buildGoProgram(gopath, packageName, outputFilePath string) error {
-	cmd := exec.Command("go", "build", "-o", outputFilePath, packageName)
-	cmd.Env = goBuildEnviron(gopath)
+func buildGoProgram(modPath, outputFilePath string) error {
+	cmd := exec.Command("go", "build", "-o", outputFilePath, ".")
+	cmd.Dir = modPath
+	cmd.Env = goBuildEnviron()
 	return run(cmd)
 }
 
-func createSingleFilePackage(gopath, packageName string, fileContents []byte) error {
-	dirName := filepath.Join(gopath, "src", packageName)
-	err := os.MkdirAll(dirName, 0700)
-	if err != nil {
-		return err
-	}
-
-	return ioutil.WriteFile(filepath.Join(dirName, "main.go"), fileContents, 0600)
-}
-
-func removePackage(gopath, packageName string) error {
-	dirName := filepath.Join(gopath, "src", packageName)
-	return os.RemoveAll(dirName)
-}
-
-func isRepoRoot(path string) bool {
-	_, err := ioutil.ReadDir(filepath.Join(path, ".git"))
-	return (err == nil) && (filepath.Base(path) == "cni")
-}
-
-func LocateCurrentGitRepo() (string, error) {
-	dir, err := os.Getwd()
-	if err != nil {
-		return "", err
-	}
-
-	for i := 0; i < 5; i++ {
-		if isRepoRoot(dir) {
-			return dir, nil
-		}
-
-		dir, err = filepath.Abs(filepath.Dir(dir))
-		if err != nil {
-			return "", fmt.Errorf("abs(dir(%q)): %s", dir, err)
-		}
-	}
-
-	return "", fmt.Errorf("unable to find cni repo root, landed at %q", dir)
-}
-
-func gitCloneThisRepo(cloneDestination string) error {
-	err := os.MkdirAll(cloneDestination, 0700)
-	if err != nil {
-		return err
-	}
-
-	currentGitRepo, err := LocateCurrentGitRepo()
-	if err != nil {
-		return err
-	}
-
-	return run(exec.Command("git", "clone", currentGitRepo, cloneDestination))
+func modInit(path, name string) error {
+	cmd := exec.Command("go", "mod", "init", name)
```

Comment:
it's failing complaining about `:` character 

```
    running "go mod init C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\cni-test-090855902": go: malformed import path "C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\cni-test-090855902": invalid char ':'
```

it seems the trick is replacing `:` by any other character 
https://github.com/armfazh/h2c-go-ref/pull/4#issue-389797462

Reply:
aha, `path.Base()` doesn't work on windows. Annoying.